packages/core/src/cli/build.ts

```diff
@@ -14,9 +18,31 @@ export async function build(
     },
   });
 
-  await rsbuildInstance.build({
+  const buildInstance = await rsbuildInstance.build({
     watch: options.watch,
   });
 
+  if (options?.watch) {
+    const files: string[] = [];
+
+    if (options.configFilePath) {
+      files.push(options.configFilePath);
+    }
+
+    onBeforeRestartServer(buildInstance.close);
+
+    watchFilesForRestart(files, async () => {
+      const { content: rslibConfig, filePath: configFilePath } =
+        await loadRslibConfig(options);
```

Comment:
Users can use `build` function as js api, `loadRslibConfig` can not be used here.